Sources/ProjectDescription/FileList.swift

```diff
@@ -4,15 +4,21 @@ public struct FileList: Codable, Equatable {
     /// List glob patterns.
     public let globs: [Path]
 
+    /// Relative glob patterns for excluded files.
+    public let excluding: [Path]
+
     /// Initializes the files list with the glob patterns.
     ///
-    /// - Parameter globs: Glob patterns.
-    public init(globs: [Path]) {
+    ///   - glob: Relative glob pattern.
+    ///   - excluding: Relative glob patterns for excluded files.
+    public init(globs: [Path],
+                excluding: [Path]? = nil) {
         self.globs = globs
+        self.excluding = excluding ?? []
     }
 
     public static func == (lhs: FileList, rhs: FileList) -> Bool {
-        lhs.globs == rhs.globs
+        lhs.globs == rhs.globs && lhs.excluding == rhs.globs
     }
```

Comment:
I think this entire function can be synthesized by the compiler?

Reply:
fixed by removing function

Reply:
Is the order of the globs supposed to matter in `FileList`s? If not, the synthesized `==` function won't always be right as `[a,b] != [b,a]`.

Reply:
my pov - if order is not important, we should use set instead of an array. as we use array here and in the similar classes, we should stay on the order-is-important and equal func too